TFNetworkLayer.py

```diff
@@ -1045,16 +1047,34 @@ def batch_norm(self, data,
       https://github.com/deepmind/sonnet/blob/master/sonnet/python/modules/batch_norm.py
     """
     with tf.variable_scope("batch_norm"):
+      import numpy as np
       if masked_time:
+        assert not use_frequency
+        # Default case
         x = data.get_placeholder_flattened(keep_dims=True)
         mean, variance = tf.nn.moments(x, axes=[0], keep_dims=True)
+        variables_shape = data.get_bc_spatial_batch_shape()
+        squeeze_variables = False
```

Comment:
Don't make the code more complicated. Why do you have `squeeze_variables = False` in the one case and `squeeze_variables = True` in the other case? Make it consistent.

Reply:
(However, that also introduces some incompatibility to older models. Make sure that this is catched and handled correctly.)

Reply:
(But you anyway have that incompatibility already with this change for the `not masked_time` case.)